src/miral/application_selector.cpp

```diff
@@ -204,6 +213,7 @@ auto ApplicationSelector::advance(bool reverse, bool within_app) -> Window
     auto it = find(selected);
 
     std::optional<Window> next_window = std::nullopt;
+    auto should_continue = true;
```

Comment:
Pretty much the same logic, but I think this is a bit clearer. Not sure if I should revert or keep these changes..